x/ccv/provider/keeper/partial_set_security_test.go

```diff
@@ -112,3 +112,28 @@ func TestHandleOptOut(t *testing.T) {
 	require.NoError(t, err)
 	require.False(t, providerKeeper.IsToBeOptedOut(ctx, "chainID", providerAddr))
 }
+
+func TestHandleSetConsumerCommissionRate(t *testing.T) {
+	providerKeeper, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	defer ctrl.Finish()
+
+	providerAddr := types.NewProviderConsAddress([]byte("providerAddr"))
+
+	// trying to set a commission rate to a unknown consumer chain
+	require.Error(t, providerKeeper.HandleSetConsumerCommissionRate(ctx, "unknownChainID", providerAddr, sdk.ZeroDec()))
+
+	// setup a pending consumer chain
+	chainID := "pendingChainID"
+	providerKeeper.SetPendingConsumerAdditionProp(ctx, &types.ConsumerAdditionProposal{ChainId: chainID})
+
+	// check that there's no commission rate set for the validator yet
+	_, found := providerKeeper.GetConsumerCommissionRate(ctx, chainID, providerAddr)
+	require.False(t, found)
+
+	require.NoError(t, providerKeeper.HandleSetConsumerCommissionRate(ctx, chainID, providerAddr, sdk.OneDec()))
```

Comment:
Maybe also have two additional checks for commission rates less than 0 and greater than 1 to avoid possible regressions.

Reply:
It's assumed that the commission rate is already in the [0, 1] range in the `HandleSetConsumerCommissionRate` scope since the range is validated in the `ValidateBasic` method of `MsgSetConsumerCommissionRate`.